tests/unit/test_spaces.cpp

```diff
@@ -315,36 +328,50 @@ TEST_P(FP32SpacesOptimizationTest, FP32InnerProductTest) {
         v2[i] = (float)(i + 1.5);
     }
 
+    auto expected_alignment = [](size_t reg_bit_size, size_t dim) {
+        size_t elements_in_reg = reg_bit_size / sizeof(float) / 8;
+        return (dim % elements_in_reg == 0) ? elements_in_reg * sizeof(float) : 0;
+    };
```

Comment:
The alignment is actually independent of the type, no? 
`alignment = reg_bit_size / 8;`

Reply:
The alignment itself yes, but we only align if the `dim * sizeof(element)` is a multiplication of the register byte size

Reply:
sure, i just meant that there is no need for a dedicated lambda for each type